### PR TITLE
Create a Sidekiq job to pull in the OGC XLS/CSV file#68646

### PR DESCRIPTION
**Pull request by holdenhinkle**
## Summary
Resolves ticket [https://github.com/department-of-veterans-affairs/va.gov-team/issues/68646](https://github.com/department-of-veterans-affairs/va.gov-team/issues/68646)

- This PR introduces 1) a new Sidekiq job to import and parse the XLSX file provided by Randy Trexler (AKA _The Trexler File_) and 2) a new Sidekiq job to update address and email fields in `veteran_representative` records and address fields in `veteran_organizations` records with data from _The Trexler File_.
- The need for this update stems from the requirement to provide geo search capabilities for representatives and organizations based on their updated address fields for the new "Find an Accredited Representative" search feature.
- The solution involves batching the `UpdateAddresses` jobs to Sidekiq, ensuring minimal impact on vets-api's performance. Additionally, the jobs run in the middle of the night.
- I am part of the team responsible for maintaining this component.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#68646

## Testing done
- Rspec tests are included
- Local testing as been done to verify that the `XlsxFileFetcher` and `QueueAddressUpdates` classes work as expected by pulling the XLSX file that we're going to use in production from a temporarily repo that I created in my GitHub account for local development purposes.
- The `UpdateAddresses` job hasn't been tested locally because the class validates addresses using the Lighthouse Address Validation API that can only be hit using approved VA IP addresses. As soon as this is merged, I'll test this job in staging.
- Additional Rspec tests to cover edge cases will be added for the `QueueAddressUpdates` job once I verify that everything is working in Staging. (Current coverage is > 96%.)

## Screenshots
_Note: Optional_

## What areas of the site does it impact?

- This change impacts the backend data handling for veteran representatives and organizations, particularly in how their address and email information is updated and maintained. It does not directly affect other areas of the site.

## Acceptance criteria

- [x] Implemented a new Sidekiq job for data import and parsing.
- [x] Implemented a new Sidekiq job to batch update addresses for Attorneys, Agents, and Organizations
- [x] No errors or warnings in the console.
- [x] Appropriate logging of events.
- [ ] ~Updated documentation with relevant information.~
- [x] No sensitive information is captured in logs or hardcoded in specs.
- [ ] ~Monitoring and alerting are set up for the new feature.~
- [ ] ~Authentication routes are verified in a local build.~
- [x] Added screenshots of the feature (if applicable).

## Requested Feedback

I would appreciate feedback on the efficiency and reliability of the data parsing logic implemented in the Sidekiq job. Any suggestions for improving data handling and error management in this context would be valuable.

### CODE REVIEW

File: modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
+
+require 'sidekiq'
+require 'sentry_logging'
+require 'roo'
+
+module RepOrgAddresses
+  # A Sidekiq job class for processing address and email updates from an Excel file.
```
Comment (jvcAdHoc) [inline modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb:8]: Just curious, do you know if there are automatic retries setup by default?

File: modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+      process_file(file_content)
+    rescue => e
+      log_error("Error in file fetching process: #{e.message}")
+    end
+
+    private
+
+    # Processes the file content by opening the spreadsheet and processing each sheet.
+    # @param file_content [String] The content of the file to be processed.
+    def process_file(file_content)
```
Comment (jvcAdHoc) [inline modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb:35]: I think a lot of this file parsing would be better served in its own class independent from the job

File: modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb (hunk 1 above)
Comment (holdenhinkle) [reply modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb:8]: Yes. Those cannot be disabled for Batches.

File: modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb (hunk 2 above)
Comment (holdenhinkle) [reply modules/veteran/app/sidekiq/rep_org_addresses/queue_address_updates.rb:35]: I agree with that - good call. Let's get this merged to verify that everything works in staging. I can circle back and refactor this later. My priority right now is to verify this in staging.

File: modules/veteran/spec/sidekiq/rep_org_addresses/queue_address_updates_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        expect_any_instance_of(SentryLogging).to receive(:log_message_to_sentry).with(
+          'QueueAddressUpdates error: Failed to fetch file or file content is empty', :error
+        )
+        subject.perform
+        expect(RepOrgAddresses::UpdateAddresses.jobs).to be_empty
+      end
+    end
+  end
+
+  describe '#process_file' do
```
Comment (jvcAdHoc) [inline modules/veteran/spec/sidekiq/rep_org_addresses/queue_address_updates_spec.rb:38]: I usually try to avoid testing private methods directly and focus on behavior. There's a fair amount of implementation detail in these specs. I think the code and testing would be cleaner with the parsing separated out from the job class.

File: config/settings.yml [hunk 4]
```diff
@@ -1686,3 +1686,6 @@ avs:
 brd:
   api_key: 'fake_key'
   base_name: 'https://something.fake.va.gov'
+
+xlsx_file_fetcher:
+  github_access_token: <%= ENV['GITHUB_ACCESS_TOKEN'] %>
```
[the remark this answers is not in the document]
Comment (holdenhinkle) [reply config/settings.yml:1691]: I'm working on this as we speak. This is the last piece that needs to be solved.

**Review by jvcAdHoc — APPROVED** [review]

**Review by RachalCassity — APPROVED** [review]
